Sources/Smithy/RequestMessage.swift

```diff
@@ -6,7 +6,7 @@
 //
 
 /// Message that is sent from client to service.
-public protocol RequestMessage {
+public protocol RequestMessage: Sendable {
```

Comment:
`Sendable` is added to request & response types so they can be safely passed through async contexts.

(`HTTPRequest` and `HTTPResponse` are both already `Sendable` so no implementation change needed for this.)